cluster-autoscaler/simulator/predicatechecker/schedulerbased.go

```diff
@@ -127,10 +152,16 @@ func (p *SchedulerBasedPredicateChecker) FitsAnyNodeMatching(clusterSnapshot clu
 		}
 
 		filterStatus := p.framework.RunFilterPlugins(context.TODO(), state, pod, nodeInfo)
-		if filterStatus.IsSuccess() {
-			p.lastIndex = (p.lastIndex + i + 1) % len(nodeInfosList)
-			return nodeInfo.Node().Name, nil
+		if !filterStatus.IsSuccess() {
+			continue
 		}
+
+		if err := p.runExtendersForNode(pod, nodeInfo); err != nil {
```

Comment:
Can we not ignore `err` here? I think it has some useful information when not `nil`.

Reply:
Fair enough. I will think something around this. Thanks for the comment.